src/app/pages/activities/club-activity/club-activity.component.ts

```diff
@@ -109,19 +136,138 @@ export class ClubActivityComponent implements OnInit {
+    const group: Group<Activity> = new Group();
+    group.id = this.eventClubInfoCreating.id;
+    this.eventClub.group = group;
+    this.clubService.addEventClub(this.eventClub).subscribe((response: ResponseUserDTO) => {
+      if (response.status_code === 200) {
+        this.dismiss();
+        swal('Chúc Mừng!', 'Đã tạo thành công!', 'success');
+      } else if (response.status_code === 903 && response.message === 'name does not allow null') {
+        swal('Thông báo!', 'Tên hoạt động không được để trống!', 'error');
+      } else if (response.status_code === 903 && response.message === 'begin date cannot null') {
+        swal('Thông báo!', 'Ngày bắt đầu không được để trống!', 'error');
+      } else if (response.status_code === 903 && response.message === 'end date cannot null') {
+        swal('Thông báo!', 'Ngày kết thúc không được để trống!', 'error');
+      } else if (response.status_code === 903 && response.message === 'list of participants cannot null') {
+        swal('Thông báo!', 'Danh sách người tham gia không được để trống!', 'error');
+      } else if (response.status_code === 932 && response.message === 'begin date is not after end date') {
+        swal('Thông báo!', 'Thời gian bắt đầu không được sau thời gian kết thúc!', 'error');
+      } else if (response.status_code === 900 && response.message === 'not find group') {
+        swal('Thông báo!', 'Hoạt động không tồn tại!', 'error');
+      } else if (response.status_code === 903 && response.message === 'user type cannot null') {
+        swal('Thông báo!', 'Loại user không được để trống!', 'error');
+      } else if (response.status_code === 900 && response.message === 'member type does not exist') {
+        swal('Thông báo!', 'Loại member không tồn tại!', 'error');
+      } else if (response.status_code === 900 && response.message === 'user does not exist') {
+        swal('Thông báo!', 'User không tồn tại!', 'error');
+      }
+    });
 
   }
 
+  onUpdate(update: any) {
+    const group: Group<Activity> = new Group();
+    group.id = this.eventClubInfoCreated.group.id;
+    this.eventClub.group = group;
+    this.eventClub.name = this.eventName;
+    this.eventClub.address = this.eventAddress;
+    this.eventClub.beginDate = this.convertNgbDateStructToString(this.startDate) + ' '
+      + this.convertNgbtimeStructToString(this.startTime);
+    this.eventClub.endDate = this.convertNgbDateStructToString(this.endDate) + ' '
+      + this.convertNgbtimeStructToString(this.endTime);
+
+    // Lấy username trong array đã chọn ở mục chọn người tham gia gán vào username
+    for (let i = 0; i < this.userClone.length; i++) {
+      const userMem: User = new User();
+      const eventUser: UserType = new UserType();
+      userMem.username = this.userClone[i].username;
+      if (userMem.username === this.hostName) {
+        eventUser.user = userMem;
+        eventUser.type = 1;
+        this.listEventUser.push(eventUser);
```

Comment:
từ line 236 này đến line 240 có thể bỏ this.listEventUser.push(eventUser); vì dòng 242 là đủ rồi

Reply:
đã sửa